pkg/controller/datadogagent/datadogagent_controller_test.go

```diff
@@ -2103,6 +2165,34 @@ func hasAllClusterLevelRbacResources(policyRules []rbacv1.PolicyRule) bool {
 	return len(clusterLevelResources) == 0
 }
 
+func hasWpaRbacs(policyRules []rbacv1.PolicyRule) bool {
+	resourceFound := false
+	groupFound := false
+	verbsFound := false
+	requiredVerbs := []string{
+		datadoghqv1alpha1.ListVerb,
+		datadoghqv1alpha1.WatchVerb,
+		datadoghqv1alpha1.GetVerb,
+	}
+	for _, policyRule := range policyRules {
+		for _, resource := range policyRule.Resources {
+			if resource == "watermarkpodautoscalers" {
+				resourceFound = true
+			}
+		}
+		for _, group := range policyRule.APIGroups {
+			if group == "datadoghq.com" {
+				groupFound = true
+			}
+		}
+		if reflect.DeepEqual(policyRule.Verbs, requiredVerbs) {
+			verbsFound = true
+		}
```

Comment:
This will make the function return false in the cases where we have more privileges than expected (i.e.: not only `get`, `list` and `watch` but also something else or even `*`.)

Reply:
yes that's true, it's a util function for a test case though I wanted to leave it as dumb as possible - I think we can keep it like this and extended if needed in other test cases in the future